### PR TITLE
[process-agent] Remove the skipping a process if it has no commandline

### PR DESCRIPTION
**Pull request by wiyu**
### What does this PR do?

This stops the dropping of a process due to missing command line which may occur due to bad parsing of the command line, a user modifying the command line to hide it.

### Describe how to test/QA your changes

1. In Linux create a program that execs programs
- `exec_hidden.c`
```c
#include <stdio.h>
#include <sys/types.h>
#include <unistd.h>
int main(int argc, char *argv[]) {
    printf("Calling hidden.c\n");
    char *args[] = {NULL};
    execv("./hidden", args);
    return 0;
}
```
- `hidden.c`
```c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>
int main(int argc, char *argv[])
{
    printf("PID of hidden.c = %d\n", getpid());

    sleep (1000);
    return 0;
}
```

2. Build the test program
```
gcc -o exec_hidden exec_hidden.c
gcc -o hidden hidden.c
```

3. Run the app
```
> ./exec_hidden
Calling hidden.c
PID of hidden.c = 495764
```
4. Check that the process is detected from the process-agent check and is available in datadog.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/process-agent-remove-proc-skip-for-missing-cmdline-3ee4cc5fd2d16652.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Process-Agent: Stop the dropping a processes when it has no commandline
```
Comment (xornivore) [inline releasenotes/notes/process-agent-remove-proc-skip-for-missing-cmdline-3ee4cc5fd2d16652.yaml:4]:
```suggestion
    Prevent reporting empty command line for processes in the Process Agent when encountering a failure to parse, report exe value instead.
```

File: pkg/process/checks/process.go [hunk 2]
```diff
@@ -537,10 +537,12 @@ func skipProcess(
 	fp *procutil.Process,
 	lastProcs map[int32]*procutil.Process,
 ) bool {
-	if len(fp.Cmdline) == 0 {
-		return true
+	cl := fp.Cmdline
+	if len(cl) == 0 {
+		cl = []string{fp.Exe}
+		log.Debugf("empty commandline, using exe:[%s] to check if the process should be skipped", cl)
```
Comment (xornivore) [inline pkg/process/checks/process.go:543]: Nit: let's add pid here.

**Review by xornivore — COMMENTED** [review]
LGTM, small nits.

**Review by alai97 — APPROVED** [review]
Looks good pending @xornivore's suggestion!

**Review by kkhor-datadog — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

**Review by just-chillin — APPROVED** [review]

**Pull request merged**